### PR TITLE
[BUG] set limit on lat lon radius for GEO

### LINKED ISSUE #2857: Endless loop on negative radius search
ref https://github.com/RedisGraph/RedisGraph/issues/2433

### PR DESCRIPTION
**Pull request by ashtul**
Before the fix we -
1. did not protect on LLAPI aginst using a negative radius or lon/lat out of range.
2. with `Dialect 2`, we didn't check `GeoFilter_Validate`.
Fix #2857

### CODE REVIEW

File: tests/pytests/test_geo.py [hunk 1]
```diff
@@ -41,6 +41,9 @@ def testGeoDistanceSimple(env):
   env.expect('FT.ADD', 'idx', 'geo3', '1', 'FIELDS', 'location', '1.23,4.55', 'hq', '1.25,4.5').ok()
   env.expect('FT.ADD', 'idx', 'geo4', '1', 'FIELDS', 'location', '1.23,4.57', 'hq', '1.25,4.5').ok()
   env.expect('FT.SEARCH', 'idx', '@location:[1.23 4.56 10 km]', 'nocontent').equal([4, 'geo1', 'geo2', 'geo3', 'geo4'])
+  env.expect('FT.SEARCH', 'idx', '@location:[1.23 4.56 -10 km]').error().contains('Invalid GeoFilter radius')
+  env.expect('FT.SEARCH', 'idx', '@location:[181 4.56 10 km]').error().contains('Invalid GeoFilter lat/lon')
+  env.expect('FT.SEARCH', 'idx', '@location:[1.23 86 10 km]').error().contains('Invalid GeoFilter lat/lon')
```
Comment (oshadmi) [inline tests/pytests/test_geo.py:46]: What is wrong here? lon=1.23 lat=86 r=10

Comment (ashtul) [reply tests/pytests/test_geo.py:46]:
```
#define GEO_LAT_MAX 85.05112878
```

File: src/geo_index.c [hunk 2]
```diff
@@ -213,7 +214,7 @@ int GeoFilter_Validate(const GeoFilter *gf, QueryError *status) {
   }
 
   // validate lat/lon
-  if (gf->lat > 90 || gf->lat < -90 || gf->lon > 180 || gf->lon < -180) {
+  if (gf->lat > GEO_LAT_MAX || gf->lat < GEO_LAT_MIN || gf->lon > GEO_LONG_MAX || gf->lon < GEO_LONG_MIN) {
```
Comment (oshadmi) [inline src/geo_index.c:217]: @ashtul Need to discuss if this is considered a breaking change (zero results Vs. error)

File: tests/cpptests/test_cpp_llapi.cpp [hunk 3]
```diff
@@ -180,13 +180,30 @@ TEST_F(LLApiTest, testAddDocumentGeoField) {
...
+  RSQNode* qn = RediSearch_CreateGeoNode(index, GEO_FIELD_NAME, 20.6543222, 0.123455, -1, RS_GEO_DISTANCE_M);
   RSResultsIterator* iter = RediSearch_GetResultsIterator(qn, index);
-  ASSERT_TRUE(iter != NULL);
+  ASSERT_FALSE(iter);
 
-  size_t len;
-  const char* id = (const char*)RediSearch_ResultsIteratorNext(iter, index, &len);
+  // lat > MAX_LAT
+  qn = RediSearch_CreateGeoNode(index, GEO_FIELD_NAME, 100, 0.123455, 10, RS_GEO_DISTANCE_M);
+  iter = RediSearch_GetResultsIterator(qn, index);  ASSERT_FALSE(iter);
```
Comment (oshadmi) [inline tests/cpptests/test_cpp_llapi.cpp:193]:
```suggestion
  iter = RediSearch_GetResultsIterator(qn, index);
```

File: tests/cpptests/test_cpp_llapi.cpp [hunk 4]
```diff
@@ -180,13 +180,30 @@ TEST_F(LLApiTest, testAddDocumentGeoField) {
...
+  ASSERT_FALSE(iter);
+
+  // lon > MAX_LON
+  qn = RediSearch_CreateGeoNode(index, GEO_FIELD_NAME, 20.6543222, 200, 10, RS_GEO_DISTANCE_M);
```
Comment (oshadmi) [inline tests/cpptests/test_cpp_llapi.cpp:197]: Maybe also here add test with
85 < `lat` < 90 ?

**Review by oshadmi — COMMENTED** [review]
Minor comments

**Review by oshadmi — APPROVED** [review]
👍🏼

**Comment by ashtul** [comment]
fixes RedisGraph/RedisGraph#2433

**Pull request merged**